docs/reference/search/search.asciidoc

```diff
@@ -51,163 +51,167 @@ include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
-  (Optional, string) A comma-separated list of fields to return as the docvalue 
+  (Optional, string) A comma-separated list of fields to return as the docvalue
   representation of a field for each hit.
-  
+
 include::{docdir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
 +
 Defaults to `open`.
-  
+
 `explain`::
-  (Optional, boolean) If `true`, returns detailed information about score 
+  (Optional, boolean) If `true`, returns detailed information about score
   computation as part of a hit. Defaults to `false`.
-  
+
 `from`::
   (Optional, integer) Defines the starting offset. Defaults to `0`.
 
 `ignore_throttled`::
-  (Optional, boolean) If `true`, concrete, expanded or aliased indices will be 
+  (Optional, boolean) If `true`, concrete, expanded or aliased indices will be
   ignored when throttled. Defaults to `false`.
 
 include::{docdir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
-  
+
 `lenient`::
-  (Optional, boolean) If `true`, format-based query failures (such as 
+  (Optional, boolean) If `true`, format-based query failures (such as
   providing text to a numeric field) will be ignored. Defaults to `false`.
-  
+
 `max_concurrent_shard_requests`::
-  (Optional, integer) Defines the number of concurrent shard requests per node 
-  this search executes concurrently. This value should be used to limit the 
-  impact of the search on the cluster in order to limit the number of concurrent 
+  (Optional, integer) Defines the number of concurrent shard requests per node
+  this search executes concurrently. This value should be used to limit the
+  impact of the search on the cluster in order to limit the number of concurrent
   shard requests. Defaults to `5`.
-  
+
 `pre_filter_shard_size`::
-  (Optional, integer) Defines a threshold that enforces a pre-filter roundtrip 
-  to prefilter search shards based on query rewriting if the number of shards 
-  the search request expands to exceeds the threshold. This filter roundtrip can 
-  limit the number of shards significantly if for instance a shard can not match 
-  any documents based on it's rewrite method ie. if date filters are mandatory 
-  to match but the shard bounds and the query are disjoint. Defaults to `128`.
+  (Optional, integer) Defines a threshold that enforces a pre-filter roundtrip
+  to prefilter search shards based on query rewriting if the number of shards
+  the search request expands to exceeds the threshold. This filter roundtrip can
+  limit the number of shards significantly if for instance a shard can not match
+  any documents based on it's rewrite method ie. if date filters are mandatory
+  to match but the shard bounds and the query are disjoint.
+  When unspecified, the pre-filter phase is executed if any of these conditions is met:
+    - The request targets more than `128` shards.
+    - The request contains read-only indices.
+    - The primary sort of the query targets an indexed field.
 
 `preference`::
```

Comment:
shall we leave the current behaviour for async search? I was under the impression that the default behaviour is now good enough also for async search and we may not need to force pre_filter_shard_size to `1` anymore. We talked about preventing users from setting such parameter manually in async search, and I think I need a refresher on these two scenarios:

1) may there be a scenario where the value must be `1` in the context of async search although these conditions are not met?
2) May it ever be necessary for users to increase it in the context of async search?

Whatever the outcome of this, we need to think about whether we want to leave the pre_filter_shard_size section/explanation in the async search docs page.

Reply:
> may there be a scenario where the value must be 1 in the context of async search although these conditions are not met?

I think so, we expect queries on time-based indices so always running the `can_match` phase makes sense to me. That's more difficult to apply to blocking queries where the threshold of `128` is used because we don't know what type of index is queried ?

> May it ever be necessary for users to increase it in the context of async search?

I agree that this part is more questionable. I am ok to remove the support for this in `async_search` if we think that it's not needed. 
We could also remove the docs for this option but leave it in the code as an opt-out.
I am fine with both options.

Reply:
> , we expect queries on time-based

That means that in most of the cases, the number of shards will be higher than 128, but I get your reasoning. Though if you don't sort on time and you query less than 128 shards on time based indices, maybe this optimization is not so useful. But I am sure it does not hurt to force 1 for async search.

I lean towards rejecting requests that try to set it. It feels like 1 is a good default for async search, and increasing it goes against the nature of async search, which is for potentially long running queries, hence you don't mind so much if there's an extra roundtrip.

Reply:
> I lean towards rejecting requests that try to set it.

Sure, no objection here ;).